### PR TITLE
fix liveslots types

### PR DESCRIPTION
**Pull request by turadg**
## Description

I noticed some typing deficiencies in https://github.com/Agoric/agoric-sdk/pull/8453

This addresses them.

### Security Considerations

--

### Scaling Considerations
--

### Documentation Considerations

--

### Testing Considerations

Some regression test added. Others "enabled" by removing suppressions.

### Upgrade Considerations

--

### CODE REVIEW

File: packages/swingset-liveslots/src/virtualObjectManager.js [hunk 1]
```diff
@@ -322,7 +322,7 @@ const insistSameCapData = (oldCD, newCD) => {
  * recursion if our returned WeakMap/WeakSet wrappers are subsequently installed
  * on globalThis.
  *
- * @returns {object} a new virtual object manager.
```
Comment (turadg) [inline packages/swingset-liveslots/src/virtualObjectManager.js:325]: `object` is `any` so this erased the type information about the function return

**Review by FUDCo — APPROVED** [review]
LGTM

**Comment by turadg** [comment]
@mergifyio requeue

**Review by erights — APPROVED** [review]
LGTM.

What are the CI failures about?

**Comment by turadg** [comment]
> What are the CI failures about?

Temporary error in the integration tests. I've kicked it off again.

**Pull request merged**